File: backend/src/main/java/com/colegio/backend/modules/teacher/application/dto/TeacherListResponse.java
package com.colegio.backend.modules.teacher.application.dto;

public record TeacherListResponse(

        Long id,
        String code,
        String email,
        String username,
        String name,
        String lastName,
        String dni,
        String birthDate,
        String phone,
        String specialty,
        String academicDegree,
        String professionalLicenseNumber,
        String university,
        Integer yearsOfExperience,
        String notes
) {
}

File: frontend/src/app/core/modules/teacher/models/teacher-list-response.ts
export interface TeacherListResponse {

  id: number;
  code: string;

  // User
  email: string;
  username: string;

  // Nombre completo
  name: string;
  lastName: string;

  // Datos personales
  firstName: string;
  middleName: string;
  paternalLastName: string;
  maternalLastName: string;

  dni: string;
  birthDate: string;
  gender: string;
  maritalStatus: string;

  // Contacto
  phone: string;
  cv:string;
  address: string;

  // Información profesional
  specialty: string;
  academicDegree: string;
  professionalLicenseNumber: string;

  // Información académica
  university: string;
  graduationDate: string;
  yearsOfExperience: number;

  // Información adicional
  notes: string;
  photo: string;
  status: string;
}